src/compiler/types.ts

```diff
@@ -157,6 +157,7 @@ namespace ts {
         SuperKeyword,
         SwitchKeyword,
         ThisKeyword,
+        ThisType, // Same as ThisKeyword, but parsed from a type context.

```

Comment:
A `ThisType` is not a keyword. 👎

Reply:
Should it contain a keyword as a child? Or should it be a leaf node, but still not a keyword?